packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -186,4 +206,81 @@ class _FloatingActionButtonState extends State<FloatingActionButton> {
 
     return result;
   }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    clearNotchMaker = Scaffold.setFloatingActionButtonNotchMakerFor(context, _notchMaker);
+  }
+
+  @override
+  void deactivate() {
+    if (clearNotchMaker != null)
+      clearNotchMaker();
+    super.deactivate();
+  }
+
+  Path _notchMaker(Rect host, Rect guest, Offset start, Offset end) {
+    assert(() {
+      if (end.dy != host.top)
+        throw new FlutterError(
+          'the floating action button\'s notch maker must only be used for a notch in the top edge.\n'
+          'end point: $end is not in the top edge of $host'
```

Comment:
i think if i saw this error message i would not be sure what "end point" means here.

Reply:
(similar comment for the next error)

Reply:
Change to "The notch's end point". Not sure if it's much better...
Though if you've seen this error it probably means that you are directly using a notch maker, so you should probably have some context about the parameters.